src/Extension/ExtensionManager.php

```diff
@@ -101,6 +109,11 @@ public function getExtensions()
 
                 $extensions->put($extension->getId(), $extension);
             }
+
```

Comment:
Instead of seeding $installedSet above in an expensive loop, why net assign it here after the loop by using the Collection pluck method:

```suggestion
$installedSet = $this->extensions->pluck('name')->toArray();
```

Reply:
I think the cost would be the same (as I'm assuming pluck does a loop under the surface), but it would be cleaner code.

Reply:
The pluck logic calls `Arr::pluck()` which is the following https://github.com/laravel/framework/blob/6.x/src/Illuminate/Support/Arr.php#L384 which does indeed do a loop. However I think using pluck is better regardless as it makes our code look a bit cleaner.

Reply:
Actually looking at this again, Arr::pluck would create an array of values. What I'm doing here uses associative arrays to create a hashset, which will have constant lookup time, compared to an array, where we'd have to iterate through that list of installed extensions, every time we check a dependency.

That being said, I do want to change it to store true instead of null as values, that seems to be convention as described by https://stackoverflow.com/a/57166943.

Reply:
You adopted the collection usage, but still defined the same variable above and use it to fill it unnecessarily above. Can you clarify what approach will be final? My preference is using a collection:

- There's no reason to use a hashset, because there's no chance that the array is assigned a duplicate value. The extensions iterated over aren't duplicate.
- We already know the extensions installed so we can simply re-use that set to retrieve the names.

Not sure this is stubborn me, feel free to let me know what you have in mind 👍

Reply:
I'm not pushing to use a hashset because of duplicate values, I want to do so for performance reasons. It takes constant time to check if an element is in a hashset, but the entire array must be iterated (although under the surface) to check if an element is in an unsorted array / collection.

Reply:
Okay that makes sense. Then I agree with your code.